arm-helper/README.md

```diff
@@ -96,14 +96,15 @@ import arm from "./arm.json"
 
 export class MyIngredient extends BaseIngredient {
 
-  public async Execute(): Promise<void> {
```

Comment:
An ingredient plugin shouldn't return anything. Since ingredients are run in parallel whenever possible we don't want to setup an environment where people are always expecting data from an ingredient execution. That's what a helper class is for.. So you can deploy a storage account via an ingredient, and use a storagehelper util class to look up things like primary key, etc.

Reply:
Cool beans

Reply:
Also, a recipe yaml format isn't really setup to be able to do anything with the results from an ingredient, which is where you'd normally want to use data like that.

Reply:
I was thinking along the lines of in Traffic Manager `parameters:
    trafficmanagername: "[traffic.get_profile()]"`. E.g. reflecting out an AppInsights key to reference a recent ingredient.

Reply:
But a helper class is just as useful

Reply:
And in parallel, threadsafe execution would be difficult to maintain

Reply:
Yeah, right thinking, just wrong approach. aiKey: "[appInsights.get_key('aiResource')]" is the better approach since it will execute that util helper in the correct context, and allows for setting dependencies to make sure aiResource is created ahead of time.

Reply:
Thanks @whilke for a productive conversation